### PR TITLE
Core/Spells: Improve teleport destination for a certain class of spel…

### LINKED ISSUE #8758: [3.3.5][Rogue]Killing spree vs Dragon bosses
Still not working part:

3) Killing spree: should be in Sindragosa stomach but he attacks Sindragosa tail and throws me far away.

Fixed part:
<details>

I have encountered a problem with Chilled to the bone debuff + Cloak of shadows + killing spree at Sindragosa 

How to reproduce the bug:
Me as Rogue Combat / Mutilate Cannot do any dps on Sindragosa Why ?

1) The bug with Chilled th the bone is : one Mutilate / one Sinister strike can stack me to 6 stacks and Two times of each skill can stack me to 8-12 stacks.
2)Cloak of Shadows: Should remove the Chill to the bone Stacks and he doesn't removes it

Video that shows every thing i write there: http://www.youtube.com/watch?v=qM3Q7BXZCng 
as you see in the video that the rogue Use Killing spree into the Sindragosa stomach and use cloak of shadows to remove his Chilled to the bone stacks and he can attack like 20 sec without getting more then 5 stacks (i get 6 in 1 sec)

Blood queen -  Swarming Shadows

How to reproduce the bug : 
as you can see in this video 1:50 https://www.youtube.com/watch?v=6rPoaHBzXSY 
The rogue use cloak of shadows and removes the swarming shadows .
Its not happens to us, even not absorb the swarming shadows with cloak of shadows on.

http://www.zeroinitiate.com/the-icecrown-citadel-blood-queen-lanathel/
Quote from the website "The final debuff is Swarming Shadows. It’s like Jaraxxus’ fire debuff except it’s purple. You can use Cloak of Shadows to remove it even before it drops a trail"

</details>

Hash: 13c8869
Tdb: 335.11.49+
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/104345-wotlk-rogue-sindragosa-chilled-to-the-bone-debuff-cloak-of-shadows-killing-spree-blood-queen-swarming-shadows?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

Edit -- Keader
Issue confirmed on rev c1b833a919fb9f3fd01bcbb2c23c32cb6e57115f too.

### PR DESCRIPTION
**Pull request by chaodhib**
**Changes proposed:**

-  Improve teleport destination for a certain class of spell. The implementation of `Spell::SelectImplicitTargetDestTargets `to be more specific.

**Target branch(es):** 3.3.5

**Issues addressed:** Closes #8758

**Tests performed:** (Does it build, tested in-game, etc.)
Tested IG. Appears to work well.

**Known issues and TODO list:**

Nothing that I'm aware of but tests should be made because this change affects a whole class of spells.

### CODE REVIEW

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -1400,12 +1400,9 @@ void Spell::SelectImplicitTargetDestTargets(SpellEffIndex effIndex, SpellImplici
...
             float angle = targetType.CalcDirectionAngle();
-            float objSize = target->GetCombatReach();
-            float dist = m_spellInfo->Effects[effIndex].CalcRadius(m_caster);
-            if (dist < objSize)
-                dist = objSize;
-            else if (targetType.GetTarget() == TARGET_DEST_TARGET_RANDOM)
-                dist = objSize + (dist - objSize) * float(rand_norm());
+            float dist = m_spellInfo->Effects[effIndex].CalcRadius(nullptr);
+            if (targetType.GetTarget() == TARGET_DEST_TARGET_RANDOM)
+                dist = dist * float(rand_norm());
```
Comment (cybrann) [inline src/server/game/Spells/Spell.cpp]: dist *= float(rand_norm());

Comment (chaodhib) [reply src/server/game/Spells/Spell.cpp]: changed. thanks.

**Comment by chaodhib** [comment]
I analysed Shadowstep. The radius of teleportation in dbc is 2.0y. However, [we can see from sniffs ](https://gist.github.com/chaodhib/13a4fa81c4a0ba38f88d92a6aa1afe0f#file-gistfile1-txt)that in practise, it's actually 4 yards. A correction is therefore needed. I assume it's an occurrence where the DBC client file does not contain the most updated values used in the server.

**Comment by Shauren** [comment]
Its still 2.0 even in legion - maybe that remaining 2 comes from somewhere else? casters or target combat reach? the mysterious bounding radius?

**Comment by chaodhib** [comment]
I don't think there is another field in the formula other than the spell radius. Because if there is one, the formula should stay valid for every spell of the same category. However, I looked at other spells within the same category (the ones with a spell effect using Spell Target 64 to 71) and on each time, the actual distance was just the spell radius. Nothing more.

Also several spells with the name 'Shadowstep' (but not the one used by rogues) also have a 4 yards teleport radius.

All these points points me torward the 'DBC data in client file not up to date' explanation.

PS: I believe Shadowstep is not in Legion anymore. It was replaced by [Shadowstrike ](http://www.wowhead.com/spell=185438/shadowstrike)(?).

**Comment by Shauren** [comment]
No, rogues still have shadowstep (with 2 charges) in legion

**Review by Shauren — APPROVED** [review]

**Comment by chaodhib** [comment]
This was not ready to be merged. because
a) Feral Charge - Cat (49376) also needs a correction on the DBC data since the spell effect is missing a radius
b) i acquired a lot of data for this class of the spell and even after a lot of diging, i couldn't figure out the formula used. What seems to be clear is that the formula doesn't use neither the target's nor the caster's combat reach.

**Comment by Shauren** [comment]
Well, there was no [WIP] tag on it so I assumed it was ready

**Pull request merged**